jax/_src/numpy/ufuncs.py

```diff
@@ -253,9 +253,54 @@ def bitwise_count(x: ArrayLike, /) -> Array:
   # Following numpy we take the absolute value and return uint8.
   return lax.population_count(abs(x)).astype('uint8')
 
-@implements(np.right_shift, module='numpy')
 @partial(jit, inline=True)
 def right_shift(x1: ArrayLike, x2: ArrayLike, /) -> Array:
+  r"""Right shift the bits of ``x1`` to the amount specified in ``x2``, element by
+  element. If x1.shape != x2.shape, they must be compatible for broadcasting to a
+  shared shape, this shared shape will also be the shape of the output. Right shifting
+  a scalar x1 by scalar x2 is equivalent to :code:`x1 // 2**x2`.
+
+  LAX-backend implementation of :func:`numpy.right_shift`.
+
+  Args:
+    x1: Input array, only accepts unsigned integer subtypes
+    x2: The amount of bits to shift each element in ``x1`` to the right, only accepts
+      integer subtypes
+
+  Returns:
+    An array-like object containing the right shifted elements of ``x1`` by the
+    amount specified in ``x2``, with the same shape as the broadcasted shape of
+    ``x1`` and ``x2``.
+
+  Example:
+    >>> x1 = jnp.array([1, 2, 4, 8])
```

Comment:
Maybe define a helper function like
```python
def print_binary(x):
  return [bin(int(val)) for val in x]
```
and use it throughout.

Then rather than having all these binary reprs in comments, you can have them in doc-tested outputs.